Replacing Wallmed1 with Oxygen_pump At main surgery

Crow-Control opened the pull request:
Replaced the Wallmed in main surgery with the anesthetics "oxygen" pump, just like in aux surgery.

Commit:
Replacing Wallmed1 with Oxygen_pump At main surgery

Replaced the Wallmed in main surgery with the anestetics "oxygen" pump,
just like in aux surgery.

Commit:
Added changelog

Added the changelog as required with mapchanges

comma commented:
I think wallmed is much more useful (inapro injectors to keep patient of a sloppy doctor alive) than this.
Can you even put those on other people, or does user have to be awake to use them? If it's latter, it's even less useful.

Crow-Control commented:
"inapro injectors to keep patient of a sloppy doctor alive"
I find the big tank system hugely anoying and quite some frequent medical players with me.

Incompetence is no excuse to add content. If you are incompetent and need this to do surgery, that is ground for a jobban anyway. In that case you shouldnt be doing it and you should have inapro with you anyway)

"Can you even put those on other people, or does user have to be awake to use them? If it's latter, it's even less useful."
I honestly don't know. In that case another edit seems needed.

Crow-Control commented:
quickly reading the oxy-pump code: you should be able to use it on other people.

chaoko99 commented:
You can clickdrag N2o pumps onto people, yes.
They also go through voidsuits for some reason, and that makes me super mad, but oh well.

Crow-Control commented:
That is a thing to look into indeed...
But well...

Crow-Control commented:
I think I can make us both content and keep the wallmed1 also.

chaoko99 commented:
Nono, I don't really care. I just think that is dumb. (The regular pumps do this too.)

Crow-Control commented:
Chaoko, what do you mean to say with "The regular pumps do this too"?

chaoko99 commented:
Regular wall mounted o2 pumps (like the ones by the escape pods) like to mysteriously apply internals through covered faces

Crow-Control commented:
ahh okey...

But what do you think, I can move the intercom and add a nanomed above the health-console at the right side of the Pomp.

The only location i can find for the intercom would be at the south corner besides the general intercom.

chaoko99 commented:
Do as you please, to be honest. I don't mind where it is, so long as they have it. as an aside, they have relatively small internal reserves of anesthetics... So a method to refill them, or a much larger resivoir would be good.

Commit:
Readded the Nanomed and moved the Medbay speaker

Due to comma wanting to keep the nanomed, I moved the (barely used)
medbay intercom. To use it now, you would have to move the freezer a
little, that wont be an issue considering it is barly used. In place I
placed a new nanomed that is now accessable from behind the health
monitor console. Which would make it more accessible for the one
assisting in surgery.

Crow-Control commented:
Okey, I read added the Nanomed as @comma  wanted and moved the Intercom to the south west corner.

Crow-Control commented:
I reread the code for the pump and the tank should be replaceable, but it also seems it already contains a normal sized tank for the anesthetics.

Commit:
Repaired area removal mistake

Mistakenly removed area at the APC, repaired with this commit

comma commented:
Thanks, it does sound like an overall improvement, tank hookup is probably slowest part of the surgery.
Now to the technical side - did you clean the map? That's a mighty huge diff on that file for few tiles changed.

Crow-Control commented:
My point exactly.
But I got your point to: To shouldn't be a choice between the two if it is possible to have both.

With the nano med above the console, the assisting doctor should have beter acces to the medicines used, I think that even helps in the cases you described: No pushing around if the other needs the nanomed.

Have a nice weekend btw ^^

comma commented:
I can't seem to find any nice guide on how to work those but basically:

Download and install 3.whatever version https://www.python.org/downloads/
Go to /tools/mapmergepy folder.
1)BEFORE editing maps run Prepare Maps.bat
2)AFTER all changes are done and saved, run Clean Maps.bat, pick maps you changed.

Since you already skipped 1) you would have to work around it. I suggest copying your current torch-4 file somewhere, reverting all changes, running Prepare Maps, overwriting file with your stashed away one, Cleaning and commiting.

Crow-Control commented:
Will do!

Commit:
cleaned Torch-4 as requested

I ran Prepare maps on a clean version, redone all the edits, put it in
the Git and prepaired the map.

Commit:
Clean slate torch 4

Ithalan commented:
I'd personally prefer that one of the intercoms remained by the surgery table. It's handy if a patient is left forgotten and immobile on the table (it could happen if the doctor meets am unexpected violent end for example) without a radio and needs to yell for help. Having both the intercoms close together in a corner also causes problems for quickly using one with :i

The nanomed could probably fit on the west wall between the air alarm and APC.

A screwdriver should also be placed in the surgery theater so doctors can easily replace the anesthetic tank in the pump once it is nearly empty.

This may be out of your intended scope for this PR, but the pump could also really use some tightening of the checks it make when attaching to people. In addition to aforementioned issue with face covering headwear, it only checks if the target is already wearing a mask at the beginning of putting it on, but not at the end.

It could also really use a prompt when clicking it directly, to prevent doctors who misclick from accidentally sedating themselves until they suffocate on the empty tank if not found. This has been a problem before with occasional poor soul who clicked the pump in aux medical.

Commit:
Edited and cleaned

This time edited and cleaned a whole clean version. This one must work.

Crow-Control commented:
@Ithalan to access that corner, the freezer needs to be moved, so placing the nanomed there wont realy be an option.

If I place the Intercom Left of the Pump, would it still work with :i from the surgeons position? (due to it being 2 tiles away instead of one)

Crow-Control commented:
@Ithalan Ahh got your point for the nanomed, missed that spot... will do!

Crow-Control commented:
@comma Got it, cleaned and well!

Commit:
Moving around as sugested by @Ithalan

@Ithalan stated that their where side effects of previous change.
It seems the westwall between APC and Airalarm is free seen from within
surgery, so moved everything around and placed the nanomed there.

Sadly no sideways nanomed available, so did the best i can with the
vertical one.

Crow-Control commented:
@Ithalan Got your suggestions into this commit

Crow-Control commented:
I also added the screwdriver, but the spawntype for the anesthetic pump is:
"spawn_type = /obj/item/weapon/tank/anesthetic", so without looking it up, that should mean it contains an standard anesthetic tank. 

While the normal O2 pump contains an emergency oxygen tank with spawn-fill (var/spawn_type = /obj/item/weapon/tank/emergency/oxygen/engi) .

Ithalan commented:
standard anesthetic tanks only contain 303 kPa, and become oxygen-poor enough to suffocate the patient at around 80 kPa, so they don't last as long as you'd expect from the size of the tank sprite.

thumbs up for getting the changes in though

Crow-Control commented:
"It could also really use a prompt when clicking it directly, to prevent doctors who misclick from accidentally sedating themselves until they suffocate on the empty tank if not found. This has been a problem before with occasional poor soul who clicked the pump in aux medical."

I suggest removing the "on click" action completely. That should be more in line with for example the IV-Drips.

"standard anesthetic tanks only contain 303 kPa, and become oxygen-poor enough to suffocate the patient at around 80 kPa, so they don't last as long as you'd expect from the size of the tank sprite."
Didnt expect them only being 303 KPa, but added the screwdriver anyway, because you will need the option to change them out what so ever. ;)

Crow-Control commented:
In addition to my suggestion of removing the "on click (with empty hand)"-action completely:
If we just add a popup, people will get annoyed with a popup creating machine at a place with a huge chance of miss-clicks.
Miss-click -> Popup, isnt nice anyway.
